### PR TITLE
[#2095] temporarily disable LJRossia crossposts

### LINKED ISSUE #2095: remove lj.rossia.org as xpost destination (while keeping it in the user tag)
https://www.dreamwidth.org/support/see_request?id=36290

From the error message, I'm going to guess that the problem is either a) lj.rossia.org doesn't have a working version of the protocol running, or b) they do but the xposter is having trouble handling it because it's a three-level domain and not a two-level domain. (I know that was a problem with adding LJR to the user tag.)

Obviously we can't fix A, but if it's B, that's something we can do something about!

### PR DESCRIPTION
**Pull request by kareila**
This removes LJRossia from the "Add External Account" form and preemptively fails any attempts to crosspost there.

It turns out that failing the job doesn't automatically notify the user, so we also make sure to send a notification to inform the user of the current situation.

Fixes #2095.

### CODE REVIEW

File: cgi-bin/DW/Worker/XPostWorker.pm [hunk 1]
```diff
@@ -62,6 +62,22 @@ sub work {
...
+    my $notify_fail = sub {
+        $sclient->insert_jobs(
+            LJ::Event::XPostFailure->new(
+                $u, $acctid, $ditemid, ( $_[0] || 'Unknown error message.' )
+            )->fire_job
+        );
+    };
+
+    # LJRossia is temporarily broken, so skip - but we do want to notify
+    if ( $acct->externalsite && $acct->externalsite->{sitename} eq 'LJRossia' ) {
```
Comment (zorkian) [inline cgi-bin/DW/Worker/XPostWorker.pm]: Wow, is this really the best way of doing this?! Hardcoding like this feels pretty sad, but I guess the original design of the xpost system didn't take into account that we might want to administratively disable things? I don't recall thinking of this use case :)

**Comment by kareila** [comment]
If we remove it from the external sites list entirely, then all the other functionality provided by DW::External::Site, like userheads, will stop working as well.

**Comment by zorkian** [comment]
We have no way to say 'this is an external site that doesn't allow crossposting'? I could have sworn that was a flag we had to turn on or off?

**Comment by kareila** [comment]
It's dependent on "servicetype" - if we update the list to say it's not an LJ-based site, that will do it, but that's (a) less obviously temporary and (b) might cause other problems.

**Comment by zorkian** [comment]
Makes sense. Thanks for thinking about it!

**Pull request merged**